### PR TITLE
Bug 1355385 - Implement PaymentAddress webAPI part

### PR DESCRIPTION
**Pull request by alphan102**
PaymentRequestManager will use "PaymentAddress::UpdateShippingAddress" to update the shippingAddress.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Just a reminder that I'm on vacation till the 25th so won't be able to review these till then. However, feel free to merge after others review and try to catch up on anything new I've missed.

File: dom/payments/PaymentRequest.cpp [hunk 1]
```diff
@@ -310,7 +329,29 @@ PaymentRequest::SetId(const nsAString& aId)
...
+                                      const nsAString& aOrganization,
+                                      const nsAString& aRecipient,
+                                      const nsAString& aPhone)
+{
+  mShippingAddress = new PaymentAddress(mWindow, aCountry, aAddressLine,
+                                        aRegion, aCity, aDependentLocality,
+                                        aPostalCode, aSortingCode, aLanguageCode,
+                                        aOrganization, aRecipient, aPhone);
+
+  // TODO : Send shippingaddresschange event
```
Comment (bentian) [inline dom/payments/PaymentRequest.cpp]: nit: s/Send/Fire

File: dom/payments/PaymentRequest.h [hunk 2]
```diff
@@ -77,6 +79,18 @@ class PaymentRequest final : public DOMEventTargetHelper
   void SetId(const nsAString& aId);
 
   already_AddRefed<PaymentAddress> GetShippingAddress() const;
+  // Update mShippingAddress and send shippingaddresschange event
```
Comment (bentian) [inline dom/payments/PaymentRequest.h]: nit: s/send/fire

File: dom/payments/PaymentRequest.cpp [hunk 3]
```diff
@@ -310,7 +329,29 @@ PaymentRequest::SetId(const nsAString& aId)
...
+                                      const nsAString& aCity,
+                                      const nsAString& aDependentLocality,
+                                      const nsAString& aPostalCode,
+                                      const nsAString& aSortingCode,
+                                      const nsAString& aLanguageCode,
+                                      const nsAString& aOrganization,
+                                      const nsAString& aRecipient,
+                                      const nsAString& aPhone)
+{
+  mShippingAddress = new PaymentAddress(mWindow, aCountry, aAddressLine,
```
Comment (bentian) [inline dom/payments/PaymentRequest.cpp]: We should be able to replace `mWindow` with `GetOwner()` (inherited from DOMEventTargetHelper, [declaration](http://searchfox.org/mozilla-central/source/dom/events/DOMEventTargetHelper.h#139)). Please open another issue/PR to revise.

Comment (alphan102) [inline dom/payments/PaymentRequest.cpp]: Create #31 for this.

**Pull request merged**